File: pset6/sentimental-readability/readability.py
# TODO

from cs50 import get_string


def main():

    text = get_string("Text: ")

    index = coleman_liau(text)

    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print("Grade ", index)

    return


def coleman_liau(text):

    letter = 0
    sentence = 0
    word = 1

    for i in text:
        if i.isalpha() == True:
            letter += 1
        elif i.isspace() == True:
            word += 1
        elif i == '.' or i == '?' or i == '!':
            sentence += 1

    L = letter * 100 / word
    S = sentence * 100 / word

    index = 0.0588 * L - 0.296 * S - 15.8

    index = round(index)

    return index


if __name__ == "__main__":
    main()

File: pset2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int coleman_liau(string sentence);

int main(void)
{
    string text = get_string("Text: ");

    int index = coleman_liau(text);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", index);
    }
}

int coleman_liau(string text)
{

    int letter = 0;
    int sentence = 0;
    int word = 1;

    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letter++;
        }
        else if (isspace(text[i]))
        {
            word++;
        }
        else if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentence++;
        }

    }

    //printf("Letters: %d\n", letter);
    //printf("Words: %d\n", word);
    //printf("Sentences: %d\n", sentence);
    float L = letter * 100.0 / word;
    float S = sentence * 100.0 / word;

    float index = 0.0588 * L - 0.296 * S - 15.8;

    //printf("Index: %f\n", index);
    index = round(index);

    return index;
}